### ISSUE TITLE
[REVIEW]: PyLithics: A Python package for stone tool analysis

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3738]
**Submitting author:** @JasonGellis (<a href="http://orcid.org/0000-0002-9929-789X">Jason Gellis</a>)
**Repository:** <a href="https://github.com/alan-turing-institute/Palaeoanalytics" target ="_blank">https://github.com/alan-turing-institute/Palaeoanalytics</a>
**Version:** v1.0.0
**Editor:** @Nikoleta-v3
**Reviewer:** @MichaelHoltonPrice , @steko 
**Archive:** <a href="https://doi.org/10.5281/zenodo.5898149" target="_blank">10.5281/zenodo.5898149</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/90f0b6e18140f98be0e24c4facc3c904/status.svg)](https://joss.theoj.org/papers/90f0b6e18140f98be0e24c4facc3c904)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/90f0b6e18140f98be0e24c4facc3c904"><img src="https://joss.theoj.org/papers/90f0b6e18140f98be0e24c4facc3c904/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/90f0b6e18140f98be0e24c4facc3c904/status.svg)](https://joss.theoj.org/papers/90f0b6e18140f98be0e24c4facc3c904)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/alan-turing-institute/Palaeoanalytics" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@MichaelHoltonPrice  & @steko , please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @Nikoleta-v3 know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @MichaelHoltonPrice 
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/alan-turing-institute/Palaeoanalytics">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/alan-turing-institute/Palaeoanalytics">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@JasonGellis) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @steko 
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/alan-turing-institute/Palaeoanalytics">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/alan-turing-institute/Palaeoanalytics">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@JasonGellis) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @MichaelHoltonPrice , @steko  it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1151

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.09 s (179.3 files/s, 166854.6 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
SVG                              1              0              1          13109
Python                           8            496            561            861
Markdown                         4            156              0            501
TeX                              1              5              0             91
YAML                             3              4              0             36
-------------------------------------------------------------------------------
SUM:                            17            661            562          14598
-------------------------------------------------------------------------------


Statistical information for the repository '8456365535258f8c256f4453' was
gathered on 2021/09/19.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Camila Rangel Smith              3           301            301            6.68
Jason Gellis                     1           140              0            1.55
JasonGellis                      6           175              8            2.03
crangelsmith                   167          4042           2628           74.03
jasongellis                      5           708            707           15.70

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Camila Rangel Smith           5            1.7          0.7                0.00
crangelsmith               1676           41.5          3.4               11.16
jasongellis                 237           33.5          1.6               12.24
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

-  10.25080/Majora-92bf1922-00a  is OK
- 10.1038/s41586-020-2649-2 is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1109/MCSE.2007.55 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: :point_left:

**Comment by Nikoleta-v3** [comment]
👋🏼 @JasonGellis @MichaelHoltonPrice  @steko this is the review thread for the paper. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread with the JOSS requirements 🔝 As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention openjournals/joss-reviews#3738 so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 2-4 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@Nikoleta-v3 ) if you have any questions/concerns 👍🏻 😄

**Comment by steko** [comment]
@Nikoleta-v3 are the steps we have taken so far (specific issues that were addressed by the authors) the good way to move forward with the review?

**Comment by Nikoleta-v3** [comment]
Yup @steko 👍🏻 

Working your way through the checklist and opening issues if you think that the paper hasn't filled the requirements is exactly what we want.

If you feel that some points require more discussion  feel free to share your thoughts and comments here, and also feel free to share your overall thoughts about the paper.

I hope I have answered your question 😄

**Comment by whedon** [comment]
:wave: @MichaelHoltonPrice , please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @steko , please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@michaelholtonprice please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by Nikoleta-v3** [comment]
👋🏻 @MichaelHoltonPrice and @steko! please update us on how your review is going 😄

**Comment by MichaelHoltonPrice** [comment]
@steko and I have raised some issues that, speaking for myself at least, need to be addressed before finishing the review. None are major, but they do prevent me from marking some items in the checklist.

**Comment by Nikoleta-v3** [comment]
Thank you for the reply @MichaelHoltonPrice! My bad, sorry,  I didn’t see the issues 👍🏻 😄

**Comment by JasonGellis** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3738 with the following error: 
```
 Error reading bibliography file paper.bib:
(line 12, column 17):
unexpected "e"
expecting space or ","
Looks like we failed to compile the PDF

```

**Comment by Nikoleta-v3** [comment]
👋🏻 @JasonGellis `whedon` failed to compile the pdf because of an error in the `bib` file. Can you please tweak the file?

**Comment by JasonGellis** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: :point_left:

**Comment by JasonGellis** [comment]
@Nikoleta-v3 The bib file has been fixed and the PDF can be generated. I have also addressed all issues raised by @steko and @MichaelHoltonPrice and merged them into the main branch of the repo.

**Comment by JasonGellis** [comment]
@Nikoleta-v3 Hello, what is the next step please?

**Comment by Nikoleta-v3** [comment]
@JasonGellis 👋🏻  We are going to give the reviewers some time to go over your corrections. Once they are happy with how you addressed their comments and they recommend the paper for publication I will also have a final look. Then we should be good to go 👍🏻 😄

**Comment by steko** [comment]
Hi, and apologies for the delay. I still have two comments about the paper that aren't specific to any of the checks but I think would help improve the quality and clarity of the final version.

1. be more explicit not only about the input images that work well, but also about the drawing styles that don't work - in my experience a lot of lithics drawings have more details than just the outline of scars and your tool does a good job but not optimal, e.g.

![sanremo1](https://user-images.githubusercontent.com/40545/144757962-ddb032ad-3e22-471f-a303-4b34c605af8e.png)

![sanremo1_lithium_scars](https://user-images.githubusercontent.com/40545/144757996-e38ceb0b-89be-4b54-822f-f67237c0581a.png)

2. the JSON output is incredibly rich but not documented in the paper, you should at least inlcude what is now in `output_example.md` file already in the repository - and maybe give some example of how to process the JSON data for further analysis?

**Comment by JasonGellis** [comment]
@steko @MichaelHoltonPrice @Nikoleta-v3 Thank you for your feedback. While differences in drawing styles are addressed in greater detail on the README.md of the repo, we agree that it belongs in the paper too, and this has been added.

We agree that that an example output of the JSON file should be included, and a new figure has been added (Figure 5). However, we feel that further discussion of how the data should be processed and analysed should be left to the user and does not belong in this paper. Metric output data and surface identification data are included in (current) figures 2 and 3. We feel that the addition of the JSON file to these two figures will adequately inform users/readers of the type of data PyLithics will produce.

All requests have been updated and merged to the main branch

**Comment by steko** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: :point_left:

**Comment by JasonGellis** [comment]
@steko @Nikoleta-v3 Great! What are the next steps  on my end?

**Comment by Nikoleta-v3** [comment]
👋🏻 @JasonGellis. Nothing to do on your end. We are currently waiting on @MichaelHoltonPrice to also review your changes :)

**Comment by JasonGellis** [comment]
@Nikoleta-v3 thanks Nikoleta!

**Comment by Nikoleta-v3** [comment]
Hey @JasonGellis  thank you for your patience. I have reached out to the reviewer via email. I have seen that you have addressed their comments but I would like to wait a little bit more from them to confirm the changes. 

Some minor comments from me:

- **Regarding the repository.** The contributing file is excellent, but it's missing some guidelines on running the test suite. Could you please add the information. 

- **Regarding the paper.**
    - Line 61 typo `arrow` -> `arrows`

**Comment by JasonGellis** [comment]
@Nikoleta-v3 I will get right on this.

**Comment by JasonGellis** [comment]
Happy New Year @Nikoleta-v3 I have addressed the above issues and merged into main. Can I request a new editor as @MichaelHoltonPrice has not been in contact since October.

**Comment by MichaelHoltonPrice** [comment]
Apologies, the notifications are not always informative (e.g., I don't
think you realized I was waiting for updates from your team on such
things as the test script for some time). Nikoleta let me know that the
manuscript was ready for me again Dec 19th but I have been on vacation
since then (and still am). I can complete my review by the end of next week.

**Comment by Nikoleta-v3** [comment]
Thank you for the update @MichaelHoltonPrice and enjoy the rest of your holiday.

**Comment by Nikoleta-v3** [comment]
Hey @JasonGellis 👋🏻  MichaelHoltonPrice opened three last issues.  They made it clear that they recommend the article for publication and that these issues are sufficient to hold up publication of the article. 

Nevertheless, I would you like you to address issue [#167](https://github.com/alan-turing-institute/Palaeoanalytics/issues/167) and issue [#169](https://github.com/alan-turing-institute/Palaeoanalytics/issues/169). Issue 169 is similar to my comment which I can see you addressed by including information regarding Travis in the README file. Could you also include guidelines on how one can run the tests locally on their machines before pushing on GitHub?

**Comment by Nikoleta-v3** [comment]
@whedon generate pdf

**Comment by Nikoleta-v3** [comment]
@whedon check references

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.25080/Majora-92bf1922-00a is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1038/s41586-020-2649-2 is OK
- 10.7717/peerj.453 is OK
- 10.1109/MCSE.2007.55 is OK

MISSING DOIs

- 10.1093/bib/5.3.305 may be a valid DOI for title: Digital code of life : how bioinformatics is revolutionizing science, medicine, and business

INVALID DOIs

- None
```

**Comment by Nikoleta-v3** [comment]
👋🏻  @JasonGellis. I `missing DOI` that `whedon` is recommending is the DOI to a review of the book. So please ignore the message (feel free to also check).

Anyway, everything looks good to me! 

At this point could you:

- [ ] Make a  new tagged release of your software if you think that it is needed (current one is v1.0), and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by JasonGellis** [comment]
Hi @Nikoleta-v3 I'm a bit unclear.
1) The version is still v1.0 so we'll keep that.
2) What am I supposed to archive on Zenodo - a copy of the repo? A pdf of the paper? 
3) I understand this.
4) Will the zenodo DOI be different from the one provided by JOSS?

**Comment by Nikoleta-v3** [comment]
1. 👍🏻 
2. Please archive the repository! In case you are not familiar, Zenodo and Github work quite nicely together ([Referencing and citing content](https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content)). You can create an archive by selecting the repo and a tagged release. I can see that you haven't made a tagged release of the software on GitHub: [your releases](https://github.com/alan-turing-institute/Palaeoanalytics/releases). Please make one and name it `v1.0.0` then use that one to archive the software.
3.  👍🏻 
4. The Zenodo DOI will be different from the JOSS one. The Zenodo archive will appear with your paper under the `Software Archive` tab. An example paper: https://joss.theoj.org/papers/10.21105/joss.03937. 

This is to ensure that an archived version of the software at the time that it was published exists 😄 Please let me know if you have any further questions, and I would be more than happy to assist you with Zenodo etc if you come across any issues! 

p.s. regarding [Referencing and citing content](https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content). I am sure that even if you have a Zenodo account already you can link it with your GitHub account.

**Comment by JasonGellis** [comment]
Hi @Nikoleta-v3 
1) The software has been tagged v1.0.0
2) The repo/software are now archived on Zenodo
3) The metadata on Zenodo is correct
4) https://zenodo.org/badge/latestdoi/303727518

**Comment by Nikoleta-v3** [comment]
@whedon set 10.5281/zenodo.5897149 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5897149" target="_blank">10.5281/zenodo.5897149</a> is the archive.

**Comment by Nikoleta-v3** [comment]
@whedon set v1.0.0 as version

**Comment by whedon** [comment]
OK. v1.0.0 is the version.

**Comment by Nikoleta-v3** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.25080/Majora-92bf1922-00a is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1038/s41586-020-2649-2 is OK
- 10.7717/peerj.453 is OK
- 10.1109/MCSE.2007.55 is OK

MISSING DOIs

- 10.1093/bib/5.3.305 may be a valid DOI for title: Digital code of life : how bioinformatics is revolutionizing science, medicine, and business

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2904

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2904, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by Nikoleta-v3** [comment]
Please note that the missing DOI that whedon is recommending is the DOI to a review of the book.

**Comment by Nikoleta-v3** [comment]
Thank you all for your work! 🚀🙏🏻

**Comment by JasonGellis** [comment]
@Nikoleta-v3 There is  an error in the author order

**Comment by JasonGellis** [comment]
@Nikoleta-v3 While I'm waiting to sort out hte issue with Zenodo, how do I see a copy of the paper? I followed the link above and coudn't generate a PDF.

**Comment by Nikoleta-v3** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: :point_left:

**Comment by JasonGellis** [comment]
@Nikoleta-v3 Ok, thank you for this. I've sorted what's happened with Zenodo. I manually changed the metadata file through their website, not realizing it was drawing infomation from the CFF file in the repo. I had hit save and thought that is was all sorted. I will update you ASAP. The paper looks good (and author names are in correct order)! Thank you for all your help.

**Comment by Nikoleta-v3** [comment]
🤔 I am not sure which date Zenodo uses as the default - I would have thought that it would use the release date. Anyways, you can edit the upload manually if you want and you can change the date to be the same as the date of the release. Picture for example: 

<img src="https://user-images.githubusercontent.com/19708408/151009185-126eac96-7446-4542-9bcf-153bcef0b990.png" alt="drawing" width="500"/>

**Comment by Nikoleta-v3** [comment]
@JasonGellis 😄

**Comment by JasonGellis** [comment]
@Nikoleta-v3 
1) Ok, I've changed this in the fields that you've shown. I'm guessing once I hit publish on the Zenodo page that it will change everything. Is this correct?
2) I've updated the images in the paper to all be the same size. The proof you listed above doesn't show this change. I've checked it in the Whendon Paper Preview Service, and it looks correct. These correctly sized images are now merged to the main branch of the repo.

 Is there anything else I need to do?

**Comment by JasonGellis** [comment]
Hi @Nikoleta-v3 everything looks good on my end and I'm happy for things to move forward to publication. I have updated Zenodo and the publication date on it is now correct: https://zenodo.org/record/5898149

The DOI: 10.5281/zenodo.5898149

**Comment by Nikoleta-v3** [comment]
👋🏻 @JasonGellis 

I just saw: https://github.com/openjournals/joss-reviews/issues/3738#issuecomment-1021395240. 

I can see that you fix the metadata for Zenodo 🔥  and the paper is under the Creative Commons Attribution 4.0 International License (CC BY 4.0) 😄

**Comment by Nikoleta-v3** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5898149" target="_blank">10.5281/zenodo.5898149</a> is the archive.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: :point_left:

**Comment by JasonGellis** [comment]
@Nikoleta-v3 It looks great! Let publish!

**Comment by Nikoleta-v3** [comment]
🤔  I didn't quite understand the question. Do you mean where is the Zenodo information on the paper? 

<img src="https://user-images.githubusercontent.com/19708408/151196444-28f49b86-bb24-47e8-a1e8-822dd4b93c7e.png" alt="drawing" width="500"/>

**Comment by JasonGellis** [comment]
@Nikoleta-v3 That answers my question. Ready to go!

**Comment by JasonGellis** [comment]
@Nikoleta-v3  Does the archive open on your PDF?

**Comment by Nikoleta-v3** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: :point_left:

**Comment by Nikoleta-v3** [comment]
@whedon set 10.5281/zenodo.5898149 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5898149" target="_blank">10.5281/zenodo.5898149</a> is the archive.

**Comment by Nikoleta-v3** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03738/joss.03738/10.21105.joss.03738.pdf) :page_facing_up: :point_left:

**Comment by Nikoleta-v3** [comment]
👋 @openjournals/joss-eics, the paper is now ready to be accepted!

I am having an issue with the `Archive` link in the paper. It doesn't redirect you to Zenodo 🤔 I tried setting the DOI again and generating the paper but that didn't work. Could you please advice? Thank you!

**Comment by JasonGellis** [comment]
@Nikoleta-v3 Does it work if you use this?: 

https://doi.org/10.5281/zenodo.5898149

**Comment by JasonGellis** [comment]
Hi @Nikoleta-v3 any updates on this?

**Comment by Nikoleta-v3** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.25080/Majora-92bf1922-00a is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1038/s41586-020-2649-2 is OK
- 10.7717/peerj.453 is OK
- 10.1109/MCSE.2007.55 is OK

MISSING DOIs

- 10.1093/bib/5.3.305 may be a valid DOI for title: Digital code of life : how bioinformatics is revolutionizing science, medicine, and business

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2911

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2911, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by Nikoleta-v3** [comment]
Apologies for the spam! 🙈 I just checked the `Archive` url in the final proof [openjournals/joss-papers#2911](https://github.com/openjournals/joss-papers/pull/2911) and it works just fine 👍🏻 

@JasonGellis, nothing more to do on our side. Now we just wait for one of the members of the editorial board to have a final look at the  submission and accept it 🎉 Thank you again for your work.

**Comment by JasonGellis** [comment]
@Nikoleta-v3 Great news! Thanks again for all your help. It's been a great experience seeing this all come together in this (new to me) context.

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1487226587309391875) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2916
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03738](https://doi.org/10.21105/joss.03738)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@MichaelHoltonPrice , @steko – many thanks for your reviews here and to @Nikoleta-v3 for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@JasonGellis – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03738/status.svg)](https://doi.org/10.21105/joss.03738)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03738">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03738/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03738/status.svg
   :target: https://doi.org/10.21105/joss.03738
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03738/status.svg)](https://doi.org/10.21105/joss.03738)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
